heat-eq-simulation/1D_CN.py

```diff
@@ -0,0 +1,54 @@
+# Crank-Nicolson method is an implicit numerical method
+# for derving the solution to PDEs
+
+# we will use the crank-nicolson method below to simulate the 1D Heat Equation
+import numpy
+import matplotlib.pyplot as pyPlot
+
+# Global Constants
+
+lengthOfRod = 10
+maxTime = 1
+heatConductivity = 100 #100 is going the least in negtive and 1000 produces the same graph to my eye
+numPointsSpace = 600
+numPointsTime = 2000
+
+# Length Vector plotted on x-axis
+xSpace = numpy.linspace(0, lengthOfRod, numPointsSpace)
+# Time Vector plotted on y-axis
+timeSpace = numpy.linspace(0, maxTime, numPointsTime)
+
+timeStepSize = timeSpace[1] - timeSpace[0]
+spaceStepSize = xSpace[1] - xSpace[0]
+
+boundaryConditions = [numpy.sin(lengthOfRod)+numpy.sin(maxTime), 
+                    	numpy.sin(lengthOfRod)]
+intialConditions = numpy.sin(xSpace)
+
+xSpaceLength = len(xSpace)
+timeSpaceLength = len(timeSpace)
+
+
+# Empty Matrix/NestedList with zeroes
+tempMatrix = numpy.zeros((xSpaceLength, timeSpaceLength))
+tempMatrix[0,:] = boundaryConditions[0]
+tempMatrix[-1,:] = boundaryConditions[1]
+tempMatrix[:, 0] = intialConditions
+
+diffusivityConstant=(heatConductivity*timeStepSize)/2*(spaceStepSize**2)
+print(diffusivityConstant)
+
+for tau in range (1, timeSpaceLength-1):
```

Comment:
@debdal6 as we discussed, this is not the CN method

Reply:
@alphaville understood will have an updated PR up before our meeting on Tuesday

Reply:
PR updated